### PR TITLE
BAK-1247 - Script for add relationships

### PR DESCRIPTION
**Pull request by diversemix**
### Script Output
Importing document relationships...
Creating relationship 'imported_relation_10007' for ids: [3451, 3117, 3452]...
... added 3451 to relationship 7
... added 3117 to relationship 7
... added 3452 to relationship 7
Creating relationship 'imported_relation_10016' for ids: [3124, 3123]...
... added 3124 to relationship 8
... added 3123 to relationship 8
Creating relationship 'imported_relation_10032' for ids: [3199, 3198]...
... added 3199 to relationship 9
... added 3198 to relationship 9
Creating relationship 'imported_relation_10039' for ids: [3924, 3923]...
... added 3924 to relationship 10
... added 3923 to relationship 10

### document_relationship Table
 document_id | relationship_id 
-------------+-----------------
        3451 |               7
        3117 |               7
        3452 |               7
        3124 |               8
        3123 |               8
        3199 |               9
        3198 |               9
        3924 |              10
        3923 |              10

### relationship Table
 id  |  type  |          name           |              description               
-----+--------+-------------------------+----------------------------------------
   3 | SYSTEM | imported_relation_10007 | imported at 2022-11-07 20:27:16.836273
   7 | SYSTEM | imported_relation_10007 | imported at 2022-11-07 20:42:44.072278
   8 | SYSTEM | imported_relation_10016 | imported at 2022-11-07 20:42:44.077331
   9 | SYSTEM | imported_relation_10032 | imported at 2022-11-07 20:42:44.079637
  10 | SYSTEM | imported_relation_10039 | imported at 2022-11-07 20:42:44.080499

### CODE REVIEW

File: backend/scripts/import_doc_relationships/import_doc_relationships.py [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+if __name__ == "__main__":
+    print("Importing document relationships...")
+    db = SessionLocal()
+    groups = db.execute(
+        """
+    SELECT * FROM (
+        SELECT (string_to_array(import_id, '.'))[3] AS grouping_id, json_agg(id) as id_list, count(id) as len
+            FROM document
+            GROUP BY grouping_id
+        ) AS T where T.len > 1;
```
Comment (joel-wright) [inline backend/scripts/import_doc_relationships/import_doc_relationships.py:32]: :scream: (but it looks right)

File: backend/scripts/import_doc_relationships/README.md [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+1. Run `nav-stack-connect.sh` from the [navigator-scripts](https://github.com/climatepolicyradar/navigator-scripts) repository, [read the docs](https://github.com/climatepolicyradar/navigator-scripts/blob/ffe777ba9f44d4570c3d8923a7fd5071f9aa4f49/docs/nav-stack-connect.md?plain=1#L1), ensuring:
+	- you have set `AWS_PROFILE` correctly and selected the correct pulumi stack
+	- sourced the correct vars file: `source ~/.aws/${AWS_PROFILE}_vars.sh`
+
+2. From the `backend` folder in the repository run:
+```bash
+PYTHONPATH=$PWD python scripts/import_doc_relationships/import_doc_relationships.py
+```
+
+## SQL
```
Comment (joel-wright) [inline backend/scripts/import_doc_relationships/README.md:18]: Maybe some explanation too? :joy:

File: backend/scripts/import_doc_relationships/import_doc_relationships.py [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+from datetime import datetime
+from app.db.models.document import DocumentRelationship, Relationship
+from app.db.session import SessionLocal
+
+
+def create_relationship_with_docs(db, name, document_ids):
+    print(f"Creating relationship '{name}' for ids: {document_ids}...")
+
+    relationship = Relationship(
+        name=name, type="SYSTEM", description=f"imported at {datetime.now()}"
```
Comment (joel-wright) [inline backend/scripts/import_doc_relationships/import_doc_relationships.py:12]: Would the description produced by this function not be more informative as something if built like this?

```suggestion
def create_relationship_with_docs(db, cclw_group_id, document_ids):
    name = f"imported_relation_{cclw_group_id}"
    print(f"Creating relationship '{name}' for ids: {document_ids}...")

    relationship = Relationship(
        name=name,
        type="SYSTEM",
        description=f"Group of related documents imported from CCLW ID {cclw_group_id}",
```

**Review by joel-wright — APPROVED** [review]
A couple of comments, but looks great (and I learned some new SQL)

File: backend/scripts/import_doc_relationships/import_doc_relationships.py (hunk 3 above)
Comment (diversemix) [reply backend/scripts/import_doc_relationships/import_doc_relationships.py:12]: I did have it like this originally - let me pop it back - I was over tidying :smile:

Comment (diversemix) [reply backend/scripts/import_doc_relationships/import_doc_relationships.py:12]: Ah understood

**Review by joel-wright — APPROVED** [review]

**Pull request merged**